.changeset/tricky-insects-argue.md

```diff
@@ -0,0 +1,24 @@
+---
+'astro': patch
+---
+
+:warning: **BREAKING CHANGE FOR EXPERIMENTAL SESSIONS ONLY** :warning:
+
+Changes the `experimental.session` option to a boolean flag and moves session config to a top-level value. This change is to allow the new automatic session driver support. You now need to separately enable the `experimental.session` flag, and then configure the session driver using the top-level `session` key if providing manual configuration.
+
+```diff
+defineConfig({
+  // ...
+  experimental: {
+-    session: {
+-      driver: 'fs',
+-    },
++    session: true,
+  },
++  session: {
++    driver: 'fs',
++  },
+});
+```
+
+You do not need to configure the session driver if you are using an adapter that supports automatic session driver configuration. For more information, see the [experimental session docs](https://docs.astro.build/en/reference/experimental-flags/sessions/).
```

Comment:
```suggestion
You no longer need to configure a session driver if you are using an adapter that supports automatic session driver configuration and wish to use its default settings.

However, you can still manually configure additional driver options or choose a non-default driver to use with your adapter with the new top-level `session` config option. For more information, see the [experimental session docs](https://docs.astro.build/en/reference/experimental-flags/sessions/).
```
Just thought *somewhere* we could mention that there are options that someone might want to set for even the default supported driver? (If I have this right.) It's not that `session` is *only* for choosing a different driver, but for setting any options like the new `ttl` etc.